eng/CodeAnalysis.ruleset

```diff
@@ -0,0 +1,472 @@
+<RuleSet Name="Microsoft.Analyzers.ManagedCodeAnalysis" Description="Microsoft.Analyzers.ManagedCodeAnalysis" ToolsVersion="14.0">
```

Comment:
Could we use an editorconfig file instead? It [sounds like](https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2019#convert-an-existing-ruleset-file-to-editorconfig-file) rulesets are deprecated and editorconfig is preferred going forward.

Reply:
Good suggestion! I moved the rules to editorconfig